src/Link.php

```diff
@@ -784,4 +707,61 @@ public static function getIcon()
     {
         return "ti ti-link";
     }
+
+    /**
+     * Validate that the provided link/content Twig template can be parsed. May not catch issues arising from runtime data.
+     * @param string $template
+     * @return bool
+     */
+    private function validateLinkOrContent(string $template)
+    {
+        try {
+            $fields = [
+                'id' => 1,
+                'name' => '',
+                'serial' => '',
+                'otherserial' => '',
+                'locations_id' => 1,
+                'networks_id' => 1,
+                'users_id' => 1,
+                'groups_id' => 1,
+                'realname' => '',
+                'firstname' => '',
+            ];
+            $fake_item = new class extends CommonDBTM {
+            };
+            $fake_item->fields = $fields;
+            self::generateLinkContents($template, $fake_item, false);
+        } catch (Throwable) {
+            Session::addMessageAfterRedirect(
+                __('Invalid link or content'),
+                false,
+                ERROR
+            );
+            return false;
+        }
+        return true;
+    }
+
+    public function prepareInputForAdd($input)
+    {
+        if (isset($input['link']) && !$this->validateLinkOrContent($input['link'])) {
+            return false;
+        }
+        if (isset($input['data']) && !$this->validateLinkOrContent($input['data'])) {
+            return false;
+        }
+        return parent::prepareInputForAdd($input);
+    }
+
+    public function prepareInputForUpdate($input)
+    {
+        if (isset($input['link']) && !$this->validateLinkOrContent($input['link'])) {
+            return false;
+        }
+        if (isset($input['data']) && !$this->validateLinkOrContent($input['data'])) {
+            return false;
+        }
+        return parent::prepareInputForUpdate($input);
+    }
```

Comment:
I think the `Glpi\ContentTemplates\TemplateManager::validate()` could be used instead.

Reply:
I pushed a commit to use this method.